File: cinema-management/frontend/src/service/employee/employeeService.ts
import { request } from "../httpClient"
import type {
  CreateEmployeeRequest,
  Employee,
  EmployeeListItem,
  PageResponse,
  UpdateEmployeeRequest,
} from "../../types/admin"

interface EmployeeSearchParams {
  page?: number
  size?: number
  keyword?: string
  status?: string
}

export function getEmployees({
  page = 0,
  size = 10,
  keyword = "",
  status = "",
}: EmployeeSearchParams = {}): Promise<PageResponse<EmployeeListItem>> {
  const params = new URLSearchParams({
    page: String(page),
    size: String(size),
  })

  if (keyword.trim()) {
    params.set("keyword", keyword.trim())
  }
  if (status) {
    params.set("status", status)
  }

  return request<PageResponse<EmployeeListItem>>(`/api/admin/employees?${params.toString()}`)
}

export function getEmployee(id: string): Promise<Employee> {
  return request<Employee>(`/api/admin/employees/${id}`)
}

export function createEmployee(data: CreateEmployeeRequest): Promise<Employee> {
  return request<Employee>("/api/admin/employees", {
    method: "POST",
    body: JSON.stringify(data),
  })
}

export function updateEmployee(id: string, data: UpdateEmployeeRequest): Promise<Employee> {
  return request<Employee>(`/api/admin/employees/${id}`, {
    method: "PUT",
    body: JSON.stringify(data),
  })
}

export function deleteEmployee(id: number): Promise<void> {
  return request<void>(`/api/admin/employees/${id}`, {
    method: "DELETE",
  })
}

File: cinema-management/backend/src/main/java/com/cinemamanagement/service/EmployeeService.java
package com.cinemamanagement.service;

import com.cinemamanagement.request.CreateEmployeeRequest;
import com.cinemamanagement.request.UpdateEmployeeRequest;
import com.cinemamanagement.response.EmployeeDetailResponse;
import com.cinemamanagement.response.EmployeeListResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface EmployeeService {
    Page<EmployeeListResponse> getEmployees(String keyword, String status, Pageable pageable);

    EmployeeDetailResponse getEmployee(Long id);

    EmployeeDetailResponse createEmployee(CreateEmployeeRequest request);

    EmployeeDetailResponse updateEmployee(Long id, UpdateEmployeeRequest request);

    void deleteEmployee(Long id);
}
